packages/Search/src/DTK_FineSearchFunctor.hpp

```diff
@@ -0,0 +1,361 @@
+    {
+        return Intrepid2::Impl::Triangle<3>::checkPointInclusion(
+            reference_point, threshold );
+    }
+};
+
+class Triangle_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_TRI_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Triangle<6>::checkPointInclusion(
+            reference_point, threshold );
+    }
+};
+
+class Wedge_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C1_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<6>::checkPointInclusion( reference_point,
+                                                               threshold );
+    }
+};
+
+class Wedge_18
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<18>::checkPointInclusion( reference_point,
+                                                                threshold );
+    }
+};
+}
+
+namespace Functor
+{
+template <typename CellType, typename DeviceType>
```

Comment:
The thing is I can't store `_cell` and `_basis` using a pointer to the base class for 2 reasons:
1) I don't think that this will work on the GPU
2) `decltype` will not return the derived class but the base class